src/toil/leader.py

```diff
@@ -41,39 +42,29 @@ def statsAndLoggingAggregatorProcess(jobStore, stop):
     startTime = time.time()
     startClock = getTotalCpuTime()
 
-    #Start off the stats file
-    with jobStore.writeSharedFileStream("statsAndLogging.xml") as fileHandle:
-        fileHandle.write('<?xml version="1.0" ?><stats>')
-
-        #Call back function
-        def statsAndLoggingCallBackFn(fileHandle2):
-            node = ET.parse(fileHandle2).getroot()
-            nodesNamed = node.find("messages").findall
-            for message in nodesNamed("message"):
-                logger.log(int(message.attrib["level"]), "Got message from job at time: %s : %s",
+    def statsAndLoggingCallBackFn(fileHandle2):
+        stats = json.load(fileHandle2, object_hook=Expando)
+        try:
+            for message in stats.workers.log:
+                logger.log(int(message.level), "Got message from job at time: %s : %s",
                            time.strftime("%m-%d-%Y %H:%M:%S"), message.text)
-            for log in nodesNamed("log"):
-                logger.info("%s:     %s" %
-                                    tuple(log.text.split("!",1)))# the jobID is separated from log by "!"
-            ET.ElementTree(node).write(fileHandle)
-
-        #The main loop
-        timeSinceOutFileLastFlushed = time.time()
-        while True:
-            if not stop.empty(): #This is a indirect way of getting a message to
-                #the process to exit
-                jobStore.readStatsAndLogging(statsAndLoggingCallBackFn)
-                break
-            if jobStore.readStatsAndLogging(statsAndLoggingCallBackFn) == 0:
-                time.sleep(0.5) #Avoid cycling too fast
-            if time.time() - timeSinceOutFileLastFlushed > 60: #Flush the
-                #results file every minute
-                fileHandle.flush()
-                timeSinceOutFileLastFlushed = time.time()
-
-        #Finish the stats file
-        fileHandle.write("<total_time time='%s' clock='%s'/></stats>" % \
-                         (str(time.time() - startTime), str(getTotalCpuTime() - startClock)))
+        except AttributeError:
+            pass  # if there are no calls to LogToMaster, the log attribute will not exist here
```

Comment:
Any occurrence of the dot operator inside the try could cause an AttributeError but only one is to be expected. Try to narrow the try block to just the one dot operator invocation that you expect to fail.